### PR TITLE
Tidy of resource/io. Align with betaflight

### PR DESCRIPTION
**Pull request by martinbudden**

### CODE REVIEW

File: src/main/drivers/resource.h [hunk 1]
```diff
@@ -34,7 +50,7 @@ typedef enum {
     OWNER_RX_SPI,
     OWNER_VTX,
     OWNER_TOTAL_COUNT
-} resourceOwner_t;
+} resourceOwner_e;
```
Comment (digitalentity) [inline src/main/drivers/resource.h:53]: Why this change? It's a `typedef` name, not a `enum` name.

Comment (martinbudden) [reply src/main/drivers/resource.h:53]: Strictly speaking this is true. However the use of the `_e` prefix for enum types gives useful information to the reader of code and the practice is already established in iNav and betaflight.

Comment (digitalentity) [reply src/main/drivers/resource.h:53]: Yes, that's true for enums. Let's keep it like this.
